spec/integration_spec.rb

```diff
@@ -435,19 +443,30 @@
     end
 
     it 'deletes the associated record' do
-      @iphone.destroy
-      results = Product.search('iphone')
+      ipad = Product.create!(name: 'ipad', href: 'apple', tags: ['awesome', 'great battery'],
+                             description: 'Big screen')
+
+      ipad.index!(true)
+      results = Product.search('ipad')
+      expect(results.size).to eq(1)
+
+      ipad.destroy
+      results = Product.search('ipad')
       expect(results.size).to eq(0)
     end
 
-    it 'does not throw an exception if a search result isn\'t found locally' do
-      Product.without_auto_index { @palmpre.destroy }
-      expect { Product.search('pal').to_json }.not_to raise_error
-    end
+    context 'when a document cannot be found in ActiveRecord' do
+      it 'does not throw an exception' do
+        Product.index.add_documents!(@palmpre.attributes.merge(id: -1))
+        expect { Product.search('pal').to_json }.not_to raise_error
+        Product.index.delete_document!(-1)
```

Comment:
I would recommend to use the [xUnit structure](https://blog.plataformatec.com.br/2014/04/improve-your-test-readability-using-the-xunit-structure/) whenever possible

Reply:
But you can do it later :)